code/modules/clothing/suits/armor.dm

```diff
@@ -349,6 +349,17 @@
 	heat_protection = UPPER_TORSO | LOWER_TORSO | LEGS | FEET | ARMS | HANDS | HEAD
 	max_heat_protection_temperature = FIRE_IMMUNITY_MAX_TEMP_PROTECT
 
+/obj/item/clothing/suit/armor/reactive/fire/equipped(mob/user, slot)
+	..()
+	if(slot == slot_wear_suit)
+		var/enchant_ID = UID(src)
+		ADD_TRAIT(user, TRAIT_RESISTHEAT, "[enchant_ID]")
```

Comment:
Early return, no need to declare a var to store UID, and finally, UID doesn't accept args.
```suggestion
	if(slot != slot_wear_suit)
		return
	ADD_TRAIT(user, TRAIT_RESISTHEAT, "[UID()]")
```